odoo/http.py

```diff
@@ -1008,6 +1017,26 @@ def cr(self, value):
 
     _cr = cr
 
+    @property
+    def geoip(self):
+        """
+        Get the remote address geolocalisation.
+
+        When geolocalization is successful, the return value is a
+        dictionary whoose format is:
+            {'city': str, 'country_code': str, 'country_name': str,
+             'latitude': float, 'longitude': float, 'region': str,
+             'time_zone': str}
+
+        When geolocalization fails, an empty dict is returned.
+        """
+        if 'geoip' not in self.session:
+            was_dirty = self.session.is_dirty
+            self.session.geoip = (self.registry['ir.http']._geoip_resolve()
```

Comment:
`_geoip` You change the api to use the geoip on request, it is better to make the session one private